### PR TITLE
op-node: Pull in Canyon Time from superchain registry

### PR DESCRIPTION
**Pull request by trianglesphere**
**Description**

Activates Canyon via the superchain registry in the op-node.

**Tests**

Manual tests. I have confirmed that the --network flag and the --override.canyon flag work correctly (including when together).

### CODE REVIEW

**Comment by tynes** [comment]
This is a fun error

```
panic: division by zero

goroutine 4160 [running]:
math/big.nat.div({0xc001c58ba0?, 0x1?, 0x6?}, {0x0?, 0x0?, 0x0?}, {0xc001c58ba0?, 0x1?, 0x6?}, {0xc001d6a668, ...})
	math/big/natdiv.go:520 +0x32e
math/big.(*Int).QuoRem(0xc001c70b20, 0xc001c70b20, 0xc0016937e0, 0xc001693720)
	math/big/int.go:294 +0x7b
math/big.(*Int).Div(0xc001d6a668?, 0xc001c70b20?, 0xc001c70ae0?)
	math/big/int.go:305 +0x38
github.com/ethereum/go-ethereum/consensus/misc/eip1559.CalcBaseFee(0xc000a033b0, 0xc001dac000, 0x653a963f)
	github.com/ethereum/go-ethereum/consensus/misc/eip1559/eip1559.go:93 +0x547
github.com/ethereum/go-ethereum/miner.(*worker).prepareWork(0xc000412b40, 0xc000e7c790)
	github.com/ethereum/go-ethereum/miner/worker.go:987 +0x54d
github.com/ethereum/go-ethereum/miner.(*worker).generateWork(0xc000412b40, 0xc000e7c790)
	github.com/ethereum/go-ethereum/miner/worker.go:1066 +0x4f
github.com/ethereum/go-ethereum/miner.(*worker).mainLoop(0xc000412b40)
	github.com/ethereum/go-ethereum/miner/worker.go:556 +0x44c
created by github.com/ethereum/go-ethereum/miner.newWorker in goroutine 1
	github.com/ethereum/go-ethereum/miner/worker.go:294 +0x846
```

**Comment by trianglesphere** [comment]
oh, I changed some variable json names but didn't change the config values (Eip1559DenomPostCanyon -> Eip1559DenomCanyon).

File: op-node/rollup/superchain.go [hunk 1]
```diff
@@ -98,6 +98,7 @@ func LoadOPStackRollupConfig(chainID uint64) (*Config, error) {
 		L1ChainID:              new(big.Int).SetUint64(superChain.Config.L1.ChainID),
 		L2ChainID:              new(big.Int).SetUint64(chConfig.ChainID),
 		RegolithTime:           &regolithTime,
+		CanyonTime:             chConfig.CanyonTime,
```
Comment (protolambda) [inline op-node/rollup/superchain.go]: Same here, we should use the superchain config

**Comment by trianglesphere** [comment]
@tynes figured it out - the devnet is pulling in from the tip of op-geth, but this is based off of a branch of op-geth

File: op-node/rollup/superchain.go (hunk 1 above)
Comment (trianglesphere) [reply op-node/rollup/superchain.go]: fixed

**Review by protolambda — APPROVED** [review]

**Pull request merged**